### PR TITLE
issue/3083 Switch to using internally derived rather than stored question models

### PR DESCRIPTION
**Pull request by oliverfoster**
fixes https://github.com/adaptlearning/adapt_framework/issues/3083

### Changed
* Switch to using internally derived rather than stored question models

### CODE REVIEW

**Review by moloko — APPROVED** [review]
code 👓

File: js/adapt-assessmentArticleModel.js [hunk 1]
```diff
@@ -253,40 +251,31 @@ define([
...
-      for (var i = 0, l = questionComponents.length; i < l; i++) {
-        var question = questionComponents[i];
-        this.stopListening(question, 'change:_isInteractionComplete', this._onQuestionCompleted);
-      }
+      this.stopListening(this.getChildren(), 'change:_isInteractionComplete', this._onBlockCompleted);
+    },
+
+    _onBlockCompleted: function(blockModel, value) {
+      if (value === false) return;
+      var questionModel = blockModel.findDescendantModels('question')[0];
```
Comment (danielghost) [inline js/adapt-assessmentArticleModel.js:276]: Not often configured this way, but if there is more than one question in a block, the assessment won't complete due to this line.

Comment (oliverfoster) [reply js/adapt-assessmentArticleModel.js]: fixed

**Review by eleanor-heath — APPROVED** [review]

**Review by moloko — APPROVED** [review]

**Review by danielghost — APPROVED** [review]
Probably don't need to return early in line 277, but looks good.

**Review by cahirodoherty-learningpool — APPROVED** [review]
👀

**Pull request merged**